### DISCUSSION TITLE
Best practice for managing env's?

### DISCUSSION THREAD

**Discussion by areohbe** [crxjs/chrome-extension-tools#83] [Q&A] [answered]
Is there a recommended pattern for support an env specific `manifest.json`. For example, if I wanted to use a different icon for dev/beta from production. Is this kinda thing supported?

**Comment by jacksteamdev** [comment, marked as answer]
You can extend the manifest using `options.extendManifest`. This option will merge the supplied object with the manifest data.

```javascript
export default {
  ...rollupConfig,
  plugins: [
    chromeExtension({
      extendManifest: {
        icons: { ... } // Overwrite just the icons object
      }
    }),
    ...plugins
  ],
}
```

I'd like to support this better in the future! Right now the manifest file name is validated as `manifest.json`, which is probably too strict. Would be nice to support other file formats too, like TS.

**Reply by areohbe** [reply]
Ah! Not sure how I missed that. Good enough for my use case.

**Reply by jacksteamdev** [reply]
@areohbe You might be interested in #94 and the [latest beta release](https://github.com/extend-chrome/rollup-plugin-chrome-extension/releases/tag/v3.6.0-7). It adds support for manifest files in TypeScript (and therefore env's)!

**Reply by areohbe** [reply]
@jacksteamdev very cool! Thanks for all the work. It made developing a Svelte-powered Chrome Extension a real breeze. It's pretty specific to particular way in which I use Gmail, but it's always nice to see your work out in the wild if you'd like to check it out. https://github.com/areohbe/power-gmail

**Reply by jacksteamdev** [reply]
Thanks for sharing! It _is_ nice to see your work being used :+1:
